src/lib.rs

```diff
@@ -251,4 +257,91 @@ mod test {
         let data: u16 = 0b1000_0000_0000_0001;
         assert!((single_ended_rdata_to_scaled_voltage(data as i16) - -V_MAX).abs() < EPS);
     }
+
+    const DEFAULT_CONFIG: u8 = 0b0000_0000;
+    const DEFAULT_STATUS: u8 = 0b0000_0001;
+    const DEVICE_ADDRESS: u8 = 0b0000_0000;
+
+    fn new_ads1119(transactions: &[I2cTransaction]) -> Ads1119<I2cMock> {
+        let device_address = 0;
+        Ads1119::new(I2cMock::new(transactions), device_address)
+    }
+
+    fn destroy_ads1119(device: Ads1119<I2cMock>) {
+        device.destroy().done();
+    }
+
+    #[test]
+    fn can_read_config() {
+        let mut device = new_ads1119(&[I2cTransaction::write_read(
+            DEVICE_ADDRESS,
+            vec![CmdFlags::RREG | RegSelectFlags::CONFIG],
+            vec![DEFAULT_CONFIG],
+        )]);
+        assert_eq!(device.read_config().unwrap(), 0b0000_0000);
+        destroy_ads1119(device);
+    }
+
+    #[test]
+    fn can_write_config() {
+        let value = 0_u8;
+        let mut device = new_ads1119(&[I2cTransaction::write(
+            DEVICE_ADDRESS,
+            vec![CmdFlags::WREG | RegSelectFlags::CONFIG, value],
+        )]);
+        device.write_config(value).unwrap();
+        destroy_ads1119(device);
+    }
+
+    #[test]
+    fn can_read_status() {
+        let mut device = new_ads1119(&[I2cTransaction::write_read(
+            DEVICE_ADDRESS,
+            vec![CmdFlags::RREG | RegSelectFlags::STATUS],
+            vec![DEFAULT_STATUS],
+        )]);
+        assert_eq!(device.read_status().unwrap(), 1);
+        destroy_ads1119(device);
+    }
+
+    #[test]
+    fn can_reset() {
+        let mut device =
+            new_ads1119(&[I2cTransaction::write(DEVICE_ADDRESS, vec![CmdFlags::RESET])]);
+        device.reset().unwrap();
+        destroy_ads1119(device);
+    }
+
+    #[test]
+    fn can_read_input_oneshot() {
+        let input = InputSelection::AN0SingleEnded;
+        let expected_output = 16383_u16;
+        let mut device = new_ads1119(&[
+            I2cTransaction::write(
+                DEVICE_ADDRESS,
+                vec![CmdFlags::WREG | RegSelectFlags::CONFIG, input.bits()],
+            ),
+            I2cTransaction::write(DEVICE_ADDRESS, vec![CmdFlags::START_SYNC]),
+            I2cTransaction::write_read(
```

Comment:
I like that you added this so that the status register would signal "not ready" one time!